lib/vector-core/src/config/log_schema.rs

```diff
@@ -104,17 +103,17 @@ impl LogSchema {
         String::from("metadata")
     }
 
-    pub fn message_key(&self) -> &str {
-        &self.message_key
+    pub fn message_key(&self) -> Option<&OwnedValuePath> {
+        self.message_key.path.as_ref()
     }
 
     /// Returns an `OwnedTargetPath` of the message key.
     /// This parses the path and will panic if it is invalid.
     ///
     /// This should only be used where the result will either be cached,
     /// or performance isn't critical, since this requires parsing / memory allocation.
-    pub fn owned_message_path(&self) -> OwnedTargetPath {
-        parse_target_path(self.message_key()).expect("valid message key")
+    pub fn owned_legacy_message_path(&self) -> OwnedTargetPath {
+        OwnedTargetPath::event(self.message_key.clone().path.expect("valid message key"))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ I don't understand why this was renamed?